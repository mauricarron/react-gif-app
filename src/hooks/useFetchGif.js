import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGif = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((img) => {
      setState({ data: img, loading: false });
    });
  }, [category]);

  return state;
};
