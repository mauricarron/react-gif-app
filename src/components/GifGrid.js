import { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { getGifs } from "../helpers/getGifs";
import { useFetchGif } from "../hooks/useFetchGif";
import GridItem from "./GridItem";

const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGif(category);

  return (
    <>
      <ul className="gif-grid">
        {loading
          ? "buscando gifs"
          : data.map((d) => <GridItem {...d} key={d.id} />)}
      </ul>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
