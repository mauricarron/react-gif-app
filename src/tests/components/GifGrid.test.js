import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGif } from "../../hooks/useFetchGif";
jest.mock("../../hooks/useFetchGif");

describe("tests on <GifGrid />", () => {
  const category = "search";

  test("should match snapshot", () => {
    useFetchGif.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot with call from useFetchGif", () => {
    const gifs = [
      { id: "ABC", title: "Mi Busqueda", url: "https://direccion.img.jpg" },
      {
        id: "DEF",
        title: "Mi Segunda Busqueda",
        url: "https://direccion.img2.jpg",
      },
    ];
    useFetchGif.mockReturnValue({
      data: gifs,
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
});
