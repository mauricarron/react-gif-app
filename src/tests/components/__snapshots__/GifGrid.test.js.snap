// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGrid /> should match snapshot 1`] = `
<Fragment>
  <ul
    className="gif-grid"
  >
    buscando gifs
  </ul>
</Fragment>
`;

exports[`tests on <GifGrid /> should match snapshot with call from useFetchGif 1`] = `
<Fragment>
  <ul
    className="gif-grid"
  >
    buscando gifs
  </ul>
</Fragment>
`;
